File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
	"unicode"
)

func main() {
	file, err := os.Open("input/text.txt")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	wordCounts := make(map[string]int)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		// Convert to lowercase
		line = strings.ToLower(line)

		// Remove punctuation and split into words
		var sb strings.Builder
		for _, r := range line {
			if unicode.IsLetter(r) {
				sb.WriteRune(r)
			} else {
				// Treat non-letters as delimiters by adding a space
				sb.WriteRune(' ')
			}
		}

		words := strings.Fields(sb.String())
		for _, w := range words {
			if w != "" {
				wordCounts[w]++
			}
		}
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	// Convert map to slice of key-value pairs for sorting
	type kv struct {
		Word  string
		Count int
	}
	var sortedList []kv
	for word, count := range wordCounts {
		sortedList = append(sortedList, kv{Word: word, Count: count})
	}

	// Sort by count descending, then by word ascending
	sort.Slice(sortedList, func(i, j int) bool {
		if sortedList[i].Count != sortedList[j].Count {
			return sortedList[i].Count > sortedList[j].Count
		}
		return sortedList[i].Word < sortedList[j].Word
	})

	// Output results
	for _, item := range sortedList {
		fmt.Printf("%s: %d\n", item.Word, item.Count)
	}
}

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/python/002_word_frequency/solution.py
import re
from collections import Counter

def count_words():
    try:
        with open('input/text.txt', 'r', encoding='utf-8') as f:
            text = f.read()
    except FileNotFoundError:
        return

    # Normalize to lowercase
    text = text.lower()
    
    # Extract words: keep only letters, split by non-letters
    words = re.findall(r'[a-z]+', text)
    
    if not words:
        return

    # Count frequency
    counts = Counter(words)
    
    # Sort by count descending, then alphabetically ascending for ties
    sorted_words = sorted(counts.items(), key=lambda x: (-x[1], x[0]))
    
    # Output
    for word, count in sorted_words:
        print(f"{word}: {count}")

if __name__ == "__main__":
    count_words()